datumaro/components/dataset_filter.py

```diff
@@ -21,8 +22,8 @@ def encode(cls, item, categories=None):
         ET.SubElement(item_elem, 'id').text = str(item.id)
         ET.SubElement(item_elem, 'subset').text = str(item.subset)
 
-        image = item.image
-        if image is not None:
+        image = item.media
+        if image and isinstance(image, Image):
```

Comment:
```suggestion
        if isinstance(image, Image):
```
The check is excessive, `None` is not an `Image`

Reply:
Fixed